xapian-core/tests/api_sorting.cc

```diff
@@ -321,79 +321,46 @@ DEFINE_TESTCASE(sortfunctorremote1, remote) {
 }
 
 DEFINE_TESTCASE(replace_weights, backend) {
-    // open the database (in this case a simple text file
-    // we prepared earlier)
-
     Xapian::Database mydb(get_database("apitest_onedoc"));
-
     Xapian::Enquire enquire(mydb);
-
-    // make a simple query, with one word in it - "word".
     enquire.set_query(Xapian::Query("word"));
-
-    // retrieve the top ten results (we only expect one)
     Xapian::MSet mymset = enquire.get_mset(0, 10);
-
-    // assigning new weight of -1 to the document.
-    vector<double> weights {-1};
+    double new_weight = (mymset.begin()).get_weight() - 0.01;
+    double old_max_possible = mymset.get_max_possible();
+    vector<double> weights {new_weight};
     mymset.replace_weights(weights.begin(), weights.end());
     Xapian::MSetIterator i = mymset.begin();
-    // checking wether the weight was assigned correctly
-    TEST_EQUAL_DOUBLE(i.get_weight(), double(-1));
-    // checking whether the max_attained attribute was updated correctly
-    TEST_EQUAL_DOUBLE(mymset.get_max_attained(), double(-1));
-
+    TEST_EQUAL_DOUBLE(i.get_weight(), new_weight);
+    TEST_EQUAL_DOUBLE(mymset.get_max_attained(), new_weight);
+    TEST_EQUAL_DOUBLE(mymset.get_max_possible(), max(old_max_possible, mymset.get_max_possible()));
```

Comment:
This is now very odd! You should just check whether the new `max_possible` is `old_max_possible`. What you're doing here actually tests whether `mymset.get_max_possible() >= old_max_possible`, but in a rather convoluted way!